### PR TITLE
Adds Validation for duplicate departments at company locations

### LINKED ISSUE #11588: [Feature Request]: Option to enforce unique Department names
### Is your feature request related to a problem? Please describe.

Currently there doesn't appear to be an option to enforce uniqueness in Department names. This may make sense if you have multiple companies, but it introduces a problem when syncing departments through various sources, and introduces the possibility of user error, especially since the dropdowns don't show which company or location a department is registered to.

### Describe the solution you'd like

Option to enforce unique Department names.

Ideally, it would be enforced by default. If we want to maintain maximum compatibility it could be enforced by default if Department has no company or location set, or share the same company and location, with an option to always require it.

### Describe alternatives you've considered

Script which double-checks and triple-checks before creating a new department, as well as routine auditing checking for duplicate names.

### Additional context

Confirmed on the current demo and model class in master as of 7-31-22.

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description
Adds Validation to check and see if the Department Name being added to a company location doesn't already exists at said company location. This only enforces unique names for Company locations.

For example:
If Grokability San Diego has a "Developer Department". It would not allow you to add another "Developer Department" to Grokability San Diego. You would have to name it "Remote Developer Department" or "Developer Department West". avoiding generalized duplicate departments for company locations.

When trying to add a duplicate department to company location:
image

Fixes #11588 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by snipe** [comment]
Can you try to name these PRs (even WIPs) a little more specifically in the future please? TY!

I'm still kinda unclear as to what this helper method should do. If we want to enforce uniqueness, we would do that at the model level. If we want to show the company (or whatever) in the dropdown, we'd do that in the presenter. Can you describe what this is going to do?

**Comment by Godmartinz** [comment]
Sorry for my vagueness. I've updated the description, and the code is pretty much in place. But it is not functioning correctly yet.

This solution is in reference to #11588

**Comment by snipe** [comment]
Right, but why would we use a helper method here versus a custom validator (or existing validator?)

File: app/Http/Controllers/DepartmentsController.php [hunk 1]
```diff
@@ -171,9 +175,35 @@ public function update(ImageUploadRequest $request, $id)
...
 
         if ($department->save()) {
+
             return redirect()->route('departments.index')->with('success', trans('admin/departments/message.update.success'));
         }
 
         return redirect()->back()->withInput()->withErrors($department->getErrors());
     }
+
+    public function passwordSave(Request $request)
```
Comment (snipe) [inline app/Http/Controllers/DepartmentsController.php]: Why is there a password save method in the ImageUploadRequest? Passwords and images are not related AFAIK.

Comment (Godmartinz) [reply app/Http/Controllers/DepartmentsController.php]: Wtf..I dont recall doing that. fixing that now

**Pull request closed without merging**